src/transformers/models/internlm3/modular_internlm3.py

```diff
@@ -0,0 +1,229 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+import sentencepiece as spm
+from torch import nn
+
+from transformers.models.llama.modeling_llama import (
+    LlamaAttention,
+    LlamaDecoderLayer,
+    LlamaForCausalLM,
+    LlamaMLP,
+    LlamaModel,
+)
+from transformers.models.llama.tokenization_llama import LlamaTokenizer
+from transformers.tokenization_utils import AddedToken, PreTrainedTokenizer
+from transformers.utils import logging
+
+
+if TYPE_CHECKING:
+    from transformers.tokenization_utils_base import TextInput
+
+from .configuration_internlm3 import InternLM3Config
+
+
+VOCAB_FILES_NAMES = {"vocab_file": "tokenizer.model"}
+
+SPIECE_UNDERLINE = "▁"
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "internlm/internlm3-8B-chat"
+_CONFIG_FOR_DOC = "InternLM3Config"
+
+
+class InternLM3Tokenizer(LlamaTokenizer, PreTrainedTokenizer):
```

Comment:
I would be in favor off rather only having a PreTrainedTokenizerFast here! How can I help making sure we only use this? (so only a fast tokenizer)

Reply:
The current HF tokenizer doesn't fully support the features in sentencepiece, and we have adopted a new vocabulary construction method, involving multi-step construction and merging, which relies on features unsupported in HF fast tokenizer. Thus, we cannot provide a precise fast tokenizer based on the current HF tokenizer library.

In our testing, the speed of the normal tokenizer and fast tokenizer is not that large.

Reply:
> The current HF tokenizer doesn't fully support the features in sentencepiece
I don't agree with you 😅 tokenizers should be 1-1 enabler for all feature. 
But mostly, this class just wraps around and is pretty much equivalent to `LlamaTokenizer` let's remove it!

`spaces_for_interleaved_special_tokens` can be controled with the `Metaspace` argument!